server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorDriver.java

```diff
@@ -451,11 +463,17 @@ public void moveSegmentOut(final String sequenceName, final List<SegmentIdentifi
       final Committer committer
   )
   {
-    final List<String> sequenceNames;
+    final List<SegmentIdentifier> theSegments;
     synchronized (activeSegments) {
-      sequenceNames = ImmutableList.copyOf(publishPendingSegments.keySet());
+      final List<String> sequenceNames = ImmutableList.copyOf(publishPendingSegments.keySet());
+      theSegments = sequenceNames.stream()
+                                 .map(publishPendingSegments::remove)
+                                 .filter(Objects::nonNull)
+                                 .flatMap(Collection::stream)
+                                 .collect(Collectors.toList());
+      sequenceNames.forEach(activeSegments::remove);
```

Comment:
This is lambda overuse, better `activeSegments.keySet().removeAll(sequenceNames)`